docs: add permissions required by kubectl CNPG plugin (#5829)
cloudnative-pg/cloudnative-pg: docs/src/kubectl-plugin.md

popular terminal-based UI to interact with Kubernetes clusters.

See [`k9s/plugins.yml`](samples/k9s/plugins.yml) for details.

## Plugin Permissions per command

The plugin will require a different set of permissions in the Kubernetes cluster
depending on the command that is required to execute, these permissions goes from
non (only local execution) to permissions over resources and sub-resources like Pods,
PDB, PVCs, etc. these permission may go from get to patch a sub resource, this
will be detailed as follow:

| Command        | Resources Permission                                                                                                          |
|:---------------|:------------------------------------------------------------------------------------------------------------------------------|

> | backup         | clusters: get<br/>backups: create

Reviewer: `certificate` command was not listed

| promote        | clusters:get<br/>clusters/status: patch<br/>pods: get                                                                         |
| psql           | pods: get,list<br/>pods/exec: create                                                                                          |
| publication    | clusters: get<br/>pods: get,list<br/>pods/exec: create                                                                        |
| reload         | clusters: get,patch                                                                                                           |
| report         | clusters: get<br/>pods: list<br/>pods/log: get<br/>jobs: list<br/>events: list                                                |
| restart        | clusters: get,patch                                                                                                           |
| status         | clusters: get<br/>pods: list<br/>pods/exec: create<br/>pods/proxy: create<br/>PDB: list                                       |
| subscription   | clusters: get<br/>pods: get,list<br/>pods/exec: create                                                                        |

### Role example

> If we want that a user has access to only get the logs of a cluster we can assign the following role:

Reviewer: ```suggestion
If we want a user can only have access to the cluster logs, we can assign the following role:
```
